app/src/main/java/com/example/mytestproject/App.kt

```diff
@@ -1,22 +1,60 @@
 package com.example.mytestproject
 
 import android.app.Application
+import android.util.Log
+import androidx.work.Configuration
+import androidx.work.ExistingPeriodicWorkPolicy
+import androidx.work.PeriodicWorkRequestBuilder
+import androidx.work.WorkManager
 import com.example.data.di.WeatherDataModule
 import com.example.mytestproject.di.CityFilterModule
 import com.example.mytestproject.di.DaggerWeatherDataComponent
 import com.example.mytestproject.di.WeatherDataComponent
+import com.example.mytestproject.workManager.DeleteOldWeatherDataWorker
+import com.example.mytestproject.workManager.SampleWorkerFactory
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
 
 class App : Application() {
 
     lateinit var weatherDataComponent: WeatherDataComponent
 
+    @Inject
+    lateinit var sampleWorkerFactory: SampleWorkerFactory
+
     override fun onCreate() {
         super.onCreate()
         weatherDataComponent = DaggerWeatherDataComponent.builder()
             .weatherDataModule(WeatherDataModule(applicationContext))
-            .cityFilterModule(CityFilterModule(applicationContext))
             .build()
 
+        weatherDataComponent.inject(this)
+
+        WorkManager.initialize(
+            this,
+            Configuration.Builder().setWorkerFactory(sampleWorkerFactory).build()
+        )
+
+        scheduleSyncWorker()
+    }
+
+    private fun scheduleSyncWorker() {
+        val workManager by lazy { WorkManager.getInstance(this) }
```

Comment:
Lazy makes sense if this variable might be not used.  But workManager will be used in any case. So, lazy is redundant here.

Reply:
Fixed